modules/lib/core/index.d.ts

```diff
@@ -142,6 +142,318 @@ export interface ScriptValue {
+export type RequestMode = 'edit' | 'inline' | 'live' | 'preview' | 'admin';
+export type RequestParams = Record<string, string | string[]>;
+export type RequestScheme = 'http' | 'https';
+
+export type RequestCookies = Record<string, string | undefined>;
+export type ResponseCookies = Record<string, string | ComplexCookie | undefined>;
+
+export type RequestHeaders = Record<string, string | undefined>;
+export type ResponseHeaders = Record<string, string | number | (string | number)[] | undefined>;
+
+export interface DefaultRequestCookies extends RequestCookies {
+    enonic_xp_tour?: string
+    JSESSIONID?: string
+}
+
+export interface DefaultRequestHeaders extends RequestHeaders {
+    Accept?: string
+    'Accept-Charset'?: string
+    'Accept-Encoding'?: string
+    'Accept-Language'?: string
+    Authorization?: string
+    'Cache-Control'?: string
+    Connection?: string
+    'Content-Length'?: string
+    'Content-Type'?: string
+    Cookie?: string
+    Language?: string
+    Host?: string
+    'If-None-Match'?: string
+    Referer?: string
+    'sec-ch-ua'?: string
+    'sec-ch-ua-mobile'?: string
+    'sec-ch-ua-platform'?: string
+    'Sec-Fetch-Dest'?: string
+    'Sec-Fetch-Mode'?: string
+    'Sec-Fetch-Site'?: string
+    'Sec-Fetch-User'?: string
+    'Upgrade-Insecure-Requests'?: string
+    'User-Agent'?: string
+    'X-Forwarded-For'?: string
+    'X-Forwarded-Host'?: string
+    'X-Forwarded-Proto'?: string
+    'X-Forwarded-Server'?: string
+}
+
+export interface RequestConstructorParams {
+    body?: string
+    branch?: LiteralUnion<RequestBranch>
+    contentType?: string
+    contextPath?: string
+    cookies: RequestCookies
+    headers: RequestHeaders
+    host: string
+    method: LiteralUnion<RequestMethod>
+    mode: LiteralUnion<RequestMode>
+    params: RequestParams
+    path: string
+    port: number
+    rawPath: string
+    remoteAddress: string
+    repositoryId?: string
+    scheme: LiteralUnion<RequestScheme>
+    url: string
+    validTicket?: boolean
+    webSocket: boolean
+}
+
+export interface RequestInterface extends RequestConstructorParams {
+    getHeader: RequestGetHeaderFunction
+}
+
+export interface DefaultRequest extends RequestInterface {
+    cookies: DefaultRequestCookies
+    headers: DefaultRequestHeaders
+}
+
+export type Request<
+    T extends Partial<RequestInterface> = Record<string, never>
+> = StrictMergeInterfaces<DefaultRequest, T>;
+
+export type SerializableRequest<T extends RequestInterface = DefaultRequest> = Omit<
+    Partial<T>,
+    'body' | 'contextPath' | 'rawPath' | 'repositoryId' | 'webSocket'
+> & {
+    body?: unknown[] | Record<string, unknown> | boolean | number | string | null;
+};
+
+export interface DefaultResponseHeaders extends ResponseHeaders {
+    'Cache-Control'?: string
+    'Content-Encoding'?: string
+    'Content-Type'?: string
+    'Content-Security-Policy'?: string
+    'Date'?: string
```

Comment:
I don't think XP allows to set this header. it is always overridden

Reply:
Which one? Date? Cause some of the other ones I do set in lib-static

Reply:
Date is forbidden in fetch requests https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date

These are response headers though.

And Date is one of those that Chrome uses.